### ISSUE TITLE
replacement_for_connection KeyError: 'output'

### ISSUE THREAD

**Issue by innovate-invent** [galaxyproject/galaxy#11612]
Galaxy 21.01
An earlier pull of the 21.01 release branch worked, but I am now seeing the following error when running my workflow:
```
galaxy.workflow.run ERROR 2021-03-12 01:39:31,002 Failed to schedule Workflow[id=7,name=IslandCompare], problem occurred on WorkflowStep[index=4,type=subworkflow].
Traceback (most recent call last):
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 358, in replacement_for_connection
    replacement = step_outputs[output_name]
KeyError: 'output'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 195, in invoke
    incomplete_or_none = self._invoke_step(workflow_invocation_step)
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 268, in _invoke_step
    incomplete_or_none = invocation_step.workflow_step.module.execute(self.trans,
  File "/srv/galaxy/lib/galaxy/workflow/modules.py", line 509, in execute
    subworkflow_invoker = progress.subworkflow_invoker(trans, step, use_cached_job=use_cached_job)
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 481, in subworkflow_invoker
    subworkflow_progress = self.subworkflow_progress(subworkflow_invocation, step, workflow_run_config.param_map)
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 499, in subworkflow_progress
    replacement = self.replacement_for_connection(
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 363, in replacement_for_connection
    raise Exception(message)
Exception: Workflow evaluation problem - failed to find output_name output in step_outputs {}
galaxy.workflow.run ERROR 2021-03-12 01:39:31,006 Failed to execute scheduled workflow.
Traceback (most recent call last):
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 358, in replacement_for_connection
    replacement = step_outputs[output_name]
KeyError: 'output'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 82, in __invoke
    outputs = invoker.invoke()
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 195, in invoke
    incomplete_or_none = self._invoke_step(workflow_invocation_step)
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 268, in _invoke_step
    incomplete_or_none = invocation_step.workflow_step.module.execute(self.trans,
  File "/srv/galaxy/lib/galaxy/workflow/modules.py", line 509, in execute
    subworkflow_invoker = progress.subworkflow_invoker(trans, step, use_cached_job=use_cached_job)
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 481, in subworkflow_invoker
    subworkflow_progress = self.subworkflow_progress(subworkflow_invocation, step, workflow_run_config.param_map)
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 499, in subworkflow_progress
    replacement = self.replacement_for_connection(
  File "/srv/galaxy/lib/galaxy/workflow/run.py", line 363, in replacement_for_connection
    raise Exception(message)
Exception: Workflow evaluation problem - failed to find output_name output in step_outputs {}
galaxy.workflow.scheduling_manager DEBUG 2021-03-12 01:39:31,050 Workflow invocation [8] scheduled
```

I think this is simmilar to https://github.com/galaxyproject/galaxy/issues/8390

**Comment by mvdbeek** [comment]
Which commit are you on ?

**Comment by innovate-invent** [comment]
43d49993f32babed58bad4bbdd53c96e72e936f9

**Comment by innovate-invent** [comment]
I just spotted this error in the log as well:
```
galaxy.jobs DEBUG 2021-03-12 19:53:29,960 job_wrapper.finish for job 103411 executed (812.088 ms)
  File "/srv/galaxy/lib/galaxy/workflow/modules.py", line 509, in execute
    subworkflow_invoker = progress.subworkflow_invoker(trans, step, use_cached_job=use_cached_job)
galaxy.jobs DEBUG 2021-03-12 19:53:53,515 finish(): Moved /data/jobs_directory/000/103/103412/outputs/galaxy_dataset_1807206e-71f1-4080-aea5-eaaec8504304.dat to /data/database/files/000/133/dataset_133617.dat
galaxy.metadata DEBUG 2021-03-12 19:53:53,570 setting metadata externally failed for HistoryDatasetAssociation 195425: Metadata results could not be read from '/data/jobs_directory/000/103/103412/metadata/metadata_results_outfile'
galaxy.jobs INFO 2021-03-12 19:53:53,694 Collecting metrics for Job 103412 in /data/jobs_directory/000/103/103412
galaxy.jobs DEBUG 2021-03-12 19:53:53,806 job_wrapper.finish for job 103412 executed (325.694 ms)
    subworkflow_progress = self.subworkflow_progress(subworkflow_invocation, step, workflow_run_config.param_map)
    replacement = self.replacement_for_connection(
    raise Exception(message)
Exception: Workflow evaluation problem - failed to find output_name output in step_outputs {}
```

**Comment by mvdbeek** [comment]
That looks a lot like the metadata job failed, so that should be a legitimate error

**Comment by innovate-invent** [comment]
The metadata jobs always fail for k8s deployments

**Comment by innovate-invent** [comment]
Resolved by #11626
